src/librustc_borrowck/borrowck/mod.rs

```diff
@@ -830,13 +834,20 @@ impl<'a, 'tcx> BorrowckCtxt<'a, 'tcx> {
                          its contents can be safely reborrowed",
                         descr)
             }
+        };
+
+        let mut db = self.struct_span_err(span, msg);
+        if let Some(span) = field_span {
+            db.span_label(span, &"this field should be `&mut`");
```

Comment:
seems like we have to check here that the type of the field is some sort of reference, right? (hence the "over fire" case in the `ui` tests below)